build/Publish.targets

```diff
@@ -84,6 +84,7 @@
                     ManifestBuildId="$(FullNugetVersion)"
                     ManifestCommit="$(CommitHash)"
                     ManifestName="$(BuildName)"
+                    ManifestBuildData="$(Product)=$(FullNugetVersion)"
```

Comment:
These should just be `"ProductVersion=$(FullNugetVersion)"`.

The final publish step actually looks for the build name `cli`, grabs the `ProductVersion` it specifies, and knows that `SDK` is the right product name for `cli`. (I didn't think it was worthwhile to make it more dynamic than that.)

Reply:
What is the difference between "ManifestBuildId" vs. "ProductVersion"? Is it just happenstance that this is the same value in the CLI?

Reply:
Yep, usually BuildId is something like 20180102-03, but it's up to you what makes sense to see for it in the manifest. ProductVersion must be that second virtual dir segment.

You can actually remove ManifestBuildId now that ProductVersion will carry the info.

I could have changed Core-Setup and ASP.NET to pass in their product version as BuildId and called it done, but that would overload the BuildId term which is already pretty well-known in Core repos IMO. I also didn't want to get *too* specialized for each repo by grabbing BuildId for CLI and ProductVersion for core-setup and aspnet.